landoapi/hgexports.py

```diff
@@ -496,13 +501,102 @@ def check_commit_message(self, is_merge: bool = False) -> Optional[str]:
 
         return "Revision needs 'Bug N' or 'No bug' in the commit message."
 
+    def check_wpt_sync(self) -> Optional[str]:
+        """Check the WPT Sync bot has only made changes to relevant subset of the tree."""
+        if self.author != "wptsync@mozilla.com":
+            return
+
+        if not self.repo:
+            return
+
+        if self.repo.tree == "try":
+            return
```

Comment:
`try` is definitely a special case, but in some cases we need to check for `if self.repo` and in others we need to check for `if not self.repo` before looking at the value of `self.repo.tree`. For now I'd like to leave it as-is, since I expect we will need to mark hooks as "run on this repo" or "exclude on this repo" more generally once we adopt more repos and run these hooks more strictly.